README.md

```diff
@@ -16,7 +16,384 @@ SPDX-License-Identifier: MIT
+            MySchedulerModule()
+            Scheduler()
+        }
+    }
+}
+```
+
+### Task Scheduling Options
+
+The Scheduler supports various scheduling patterns using the [`Schedule`](https://swiftpackageindex.com/stanfordspezi/spezischeduler/documentation/spezischeduler/schedule) type:
+
+```swift
+// One-time task
+let onceSchedule = Schedule.once(at: Date(), duration: .tillEndOfDay)
+
+// Daily tasks
+let dailySchedule = Schedule.daily(hour: 8, minute: 30, startingAt: .today)
+
+// Weekly tasks
+let weeklySchedule = Schedule.weekly(
+    weekday: .monday, 
+    hour: 10, 
+    minute: 0, 
+    startingAt: .today
+)
+
+// Monthly tasks
+let monthlySchedule = Schedule.monthly(
+    day: 1, 
+    hour: 9, 
+    minute: 0, 
+    startingAt: .today
+)
+
+// Custom recurrence patterns
+var customRule = Calendar.RecurrenceRule.weekly(calendar: .current, end: .never)
+customRule.weekdays = [.every(.monday), .every(.wednesday), .every(.friday)]
+let customSchedule = Schedule(startingAt: .today, recurrence: customRule)
+```
+
+### Task Categories and Metadata
+
+Tasks can include categories and additional metadata for better organization and functionality:
+
+```swift
+try scheduler.createOrUpdateTask(
+    id: "medication-reminder",
+    title: "Morning Medication",
+    instructions: "Take your prescribed morning medication with water.",
+    category: .medication,
+    schedule: .daily(hour: 8, minute: 0, startingAt: .today),
+    tags: ["health", "medication", "daily"]
+) { context in
+    // Store additional metadata using the @Property macro
+    context.about = "Take your daily medication as prescribed by your healthcare provider."
+}
+```
+
+### Notifications
+
+For basic notification support, you can use the [`Notifications`](https://github.com/StanfordSpezi/SpeziNotifications) module in your app configuration:
+
+```swift
+class ExampleAppDelegate: SpeziAppDelegate {
+    override var configuration: Configuration {
+        Configuration(standard: ExampleStandard()) {
+            MySchedulerModule()
+            Scheduler()
+            Notifications()
+        }
+    }
+}
+```
+
+For advanced scheduler-specific notification features, see the [`SchedulerNotifications`](https://swiftpackageindex.com/stanfordspezi/spezischeduler/documentation/spezischeduler/schedulernotifications) module documentation.
+
+### Querying Tasks and Events
+
+You can query tasks and events using various methods:
+
+```swift
+// Query tasks for a specific date range
+let tasks = try scheduler.queryTasks(for: Date()..<Calendar.current.date(byAdding: .day, value: 7, to: Date())!)
+
+// Query events (task occurrences) for today
+let todayEvents = try scheduler.queryEvents(for: Date()..<Calendar.current.date(byAdding: .day, value: 1, to: Date())!)
+
+// Query events for a specific task
+let taskEvents = try scheduler.queryEvents(forTaskWithId: "daily-questionnaire", in: Date()..<Date().addingTimeInterval(86400))
```

Comment:
we have nice APIs for (at least some of) these date computations in SpeziFoundation; eg you can also do `scheduler.queryEvents(for: calendar.rangeOfDay(for: .now))`